### PR TITLE
[Doc] Add view-based Rewrite

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/using_starrocks/query_rewrite_with_materialized_views.md [hunk 1]
```diff
@@ -706,36 +706,282 @@ GROUP BY lo_orderkey;
...
-例如，创建以下视图：
+然而，在实际场景中，数据分析师可能会基于复杂的嵌套视图进行数据建模，这些视图无法直接展开。因此，基于这些视图创建的物化视图无法改写查询。为了改进在上述情况下的能力，从 v3.3.0 开始，StarRock 优化了基于视图的物化视图查询改写逻辑。
+
+### 基本原理
+
+在先前的查询改写逻辑中，StarRocks 会将基于视图的查询展开为针对视图基表的查询。如果展开后查询的执行计划与 SPJG 模式不匹配，物化视图将无法改写查询。
+
+为了解决这个问题，StarRocks 引入了一个新的算子 - LogicalViewScanOperator，该算子用于简化执行计划树的结构，且无需展开查询，使查询执行计划树尽量满足 SPJG 模式，从而优化查询改写。
+
+以下示例展示了一个具有聚合子查询的查询，一个建立在子查询之上的视图，基于视图展开后查询，以及建立在视图之上的物化视图：
```
Comment (evelynzhaojie) [inline docs/zh/using_starrocks/query_rewrite_with_materialized_views.md:721]: 1. 具有 -- 包含

File: docs/en/using_starrocks/query_rewrite_with_materialized_views.md [hunk 2]
```diff
@@ -710,34 +710,280 @@ As shown above, `agg_mv5` contains the data from partitions `p1` to `p7`, and th
 
 ## View-based materialized view rewrite
 
-StarRocks supports creating materialized views based on views. Subsequent queries against the views can be transparently rewritten.
+From v3.1.0 onwards, StarRocks supports creating materialized views based on views. Subsequent queries against the views can be rewritten if they are of the SPJG pattern. By default, queries against views are automatically transcribed into queries against the base tables of the views and then transparently matched and rewritten.
 
-For example, create the following views:
+However, in real-world scenarios, data analysts may perform data modeling upon complicated, nested views, which cannot be directly transcribed. As a result, materialized views created based on such views cannot rewrite queries. To improve its capability in the preceding scenario, StarRocks optimizes the view-based materialized view query rewrite logic from v3.3.0 onwards.
```
Comment (evelynzhaojie) [inline docs/en/using_starrocks/query_rewrite_with_materialized_views.md:715]: complicated --- complex

File: docs/en/using_starrocks/query_rewrite_with_materialized_views.md [hunk 3]
```diff
@@ -710,34 +710,280 @@ As shown above, `agg_mv5` contains the data from partitions `p1` to `p7`, and th
...
+
+### Fundamentals
+
+In the previous query rewrite logic, StarRocks will transcribe queries against the view into queries against the base tables of the view. Query rewrite will encounter failures if the execution plan of the transcribed query mismatches the SPJG pattern.
```
Comment (evelynzhaojie) [inline docs/en/using_starrocks/query_rewrite_with_materialized_views.md:719]: against the view --- against a view

File: docs/en/using_starrocks/query_rewrite_with_materialized_views.md [hunk 4]
```diff
@@ -710,34 +710,280 @@ As shown above, `agg_mv5` contains the data from partitions `p1` to `p7`, and th
...
+
+To solve this problem, StarRocks introduces a new operator - LogicalViewScanOperator, to simplify the structure of the execution plan tree without transcribing the query. This operator seeks to match the execution plan tree with the SPJG pattern, therefore facilitating query rewrite.
+
+The following example lists a query with an AGGREGATE sub-query, a view built upon the sub-query, the transcribed query based on the view, and the materialized built upon the view:
```
Comment (evelynzhaojie) [inline docs/en/using_starrocks/query_rewrite_with_materialized_views.md:723]: the materialized -- the materialized view

File: docs/en/using_starrocks/query_rewrite_with_materialized_views.md [hunk 5]
```diff
@@ -802,10 +1048,24 @@ For asynchronous materialized views created based on an external catalog, you ca
...
 - Materialized views defined with statements containing LIMIT, ORDER BY, UNION, EXCEPT, INTERSECT, MINUS, GROUPING SETS, WITH CUBE, or WITH ROLLUP cannot be used for query rewrite.
 - Strong consistency of query results is not guaranteed between base tables and materialized views built on external catalogs.
 - Asynchronous materialized views created on base tables in a JDBC catalog do not support query rewrite.
+
+In terms of view-based materialized view query rewrite, StarRocks currently has the following limitations:
+
+- Currently, StarRocks does not support Partition Union rewrite.
+- Query rewrite is not supported if the view contains random functions, including rand(), random(), uuid(), and sleep().
+- Query rewrite is not supported if the view contains columns with same names. You must assign different aliases for columns with the same names.
+- Views that used to create a materialized view must contain at least one column of the following data types: integer types, date types, and string types. For example, you cannot create a materialized that queries the view, because `total_cost` is a DOUBLE-type column.
```
Comment (evelynzhaojie) [inline docs/en/using_starrocks/query_rewrite_with_materialized_views.md:1064]: that used to --- that are used to

File: docs/zh/using_starrocks/query_rewrite_with_materialized_views.md [hunk 6]
```diff
@@ -706,36 +706,282 @@ GROUP BY lo_orderkey;
 
 如上所示，`agg_mv5` 包含来自分区 `p1` 到 `p7` 的数据，而分区 `p8` 的数据来源于 `lineorder`。最后，这两组数据使用 UNION 操作合并。
 
-## 基于视图构建物化视图
+## 基于视图的物化视图查询改写
```
Comment (evelynzhaojie) [inline docs/zh/using_starrocks/query_rewrite_with_materialized_views.md:709]: image
 看看异步物化视图的文档里，需不需要加 基于视图的改写 相关内容

Comment (evelynzhaojie) [reply docs/zh/using_starrocks/query_rewrite_with_materialized_views.md:709]: https://docs.starrocks.io/zh/docs/using_starrocks/Materialized_view/#%E4%BD%BF%E7%94%A8%E5%BC%82%E6%AD%A5%E7%89%A9%E5%8C%96%E8%A7%86%E5%9B%BE%E6%94%B9%E5%86%99%E5%8A%A0%E9%80%9F%E6%9F%A5%E8%AF%A2

**Review by evelynzhaojie — APPROVED** [review]

**Review by hellolilyliuyi — APPROVED** [review]

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-3.2

**Pull request merged**